### PR TITLE
fix: WP-890 - article footnote rendering in all articles not just special reports

### PR DESCRIPTION
**Pull request by yasgreen93**
https://jira.economist.com/browse/WP-890

For some odd reason when I first implemented this functionality of adding the small footnote at the bottom of articles I put it in the functionality for special reports only. This PR moves it so that it appears in all articles that need the footNote.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -69,8 +70,33 @@ const otherProps = {
   TitleComponent: ({ flyTitle, title }) => (<div className="test-title-component">test: {flyTitle} {title}</div>),
 }
 
+const moreProps = {
```
Comment (etaylor23) [inline test/index.js]: `moreProps` might be more suited to its own file and imported in

File: src/index.js [hunk 2]
```diff
@@ -330,7 +329,12 @@ export default class BlogPost extends React.Component {
       blogPostTextElements.splice(articleListPosition, 0, siblingArticlesList);
       content.splice(content.length - 1, 0, nextArticleLink);
     }
-    if (articleFootNote && printEdition) {
+  }
+
+  addArticleFootNote(content, articleFootNote) {
+    let blogPostTextElements = null;
```
Comment (umbobabo) [inline src/index.js]: This var is block scoped here and you are not returning it.
You should try to keep a more functional programming approach cause in this file this...
```
blogPostTextElements = this.filterBlogPostTextElements(content);
```
seems to be repeated a lot of times in a not really useful way.

File: test/index.js (hunk 1 above)
Comment (umbobabo) [inline test/index.js]: Let's try to give a meaning to the vars.
Or we end-up in this horrible situation:
```
 const mountComponentWithProps = mountComponent(requiredProps);
 const mountComponentWithOtherProps = mountComponent(otherProps)
 const mountComponentWithMoreProps = mountComponent(moreProps)
```
@etaylor23 I think the conf should be near to the test if there is no reusability necessity.
But we can split the tests in different files if this file start to be too complex/messed up.

Comment (umbobabo) [inline test/index.js]: @yasgreen93 Don't use a complete new config object if you just need 1 or 2 additionals props. use Object.assign to override what you need from one already existing object.

**Pull request merged**